libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -521,7 +522,7 @@ void AP_Baro::update(void)
             // update altitude calculation
             float ground_pressure = sensors[i].ground_pressure;
             if (!is_positive(ground_pressure) || isnan(ground_pressure) || isinf(ground_pressure)) {
-                sensors[i].ground_pressure = sensors[i].pressure;
```

Comment:
This may be right (most of the time) for subs, it's totally wrong for any other platform. What about subs in lakes?